File: 3090. Maximum Length Substring With Two Occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        left=0
        ans=0
        map={}
        for i,ch in enumerate(s):
            map[ch]=map.get(ch,0)+1

            while(map[ch]>2):
                map[s[left]]-=1
                left+=1
            
            ans=max(ans,i-left+1)
        return ans

File: 3090. Maximum Length Substring With Two Occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        int left=0;
        int ans=0;
        vector<int> map(26,0);
        for(int right=0;right<s.size();right++)
        {
            char ch=s[right];
            int pos=ch-'a';

            map[pos]++;

            while(map[pos]>2){
                map[s[left]-'a']--;
                left++;
            }    
            ans=max(ans,right-left+1);
        }
        return ans;
    }
};